contracts/modules/viewer/NexusViewer.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import {Multicall} from "../../abstract/Multicall.sol";
+import {IAssessment} from "../../interfaces/IAssessment.sol";
+import {IAssessmentViewer} from "../../interfaces/IAssessmentViewer.sol";
+import {IGovernance} from "../../interfaces/IGovernance.sol";
+import {INXMMaster} from "../../interfaces/INXMMaster.sol";
+import {IPooledStaking} from "../../interfaces/IPooledStaking.sol";
+import {IStakingViewer} from "../../interfaces/IStakingViewer.sol";
+import {ITokenController} from "../../interfaces/ITokenController.sol";
+
+/// @title NexusViewer Contract
+/// @notice This contract provides a unified view of system-wide data from various contracts within the Nexus Mutual platform.
+contract NexusViewer is Multicall {
+  struct LegacyPoolStake {
+    uint deposit;
+    uint reward;
+  }
+
+  struct ClaimableNxm {
+    uint governanceRewards;
+    IStakingViewer.AggregatedTokens aggregateStakingTokens;
+    IAssessmentViewer.AssessmentRewards assessmentRewards;
+    LegacyPoolStake legacyPooledStake;
+    uint v1CoverNotesAmount;
+  }
+
+  struct LockedNxm {
+    IStakingViewer.AggregatedTokens aggregatedTokens;
```

Comment:
consider only using `activeStake` here from the AggregatedTokens struct. the other 2 fields are irelevant when referring to stakedNXM